Sources/Sentry/SentryUIViewControllerPerformanceTracker.m

```diff
@@ -109,6 +120,53 @@ - (void)createTransaction:(UIViewController *)controller
     }
 }
 
+- (void)createTimeToDisplay:(UIViewController *)controller
+{
+    SentrySpan *vcSpan = [self viewControllerPerformanceSpan:controller];
+
+    if (![vcSpan isKindOfClass:[SentryTracer self]]) {
+        // Since TTID and TTFD are meant to the whole screen
+        // we will not track child view controllers
+        return;
+    }
+
+    if ([vcSpan.tracer getExtensionsOfType:SentryTimeToDisplayTracker.self].count > 0) {
+        // Already tracking time to display, not creating a new tracker.
+        // This may happen if user manually call `loadView` from a view controller more than once.
+        return;
+    }
+
+    if (self.currentTTDTracker) {
+        [self.currentTTDTracker stopWaitingFullDisplay];
+    }
+
+    SentryTimeToDisplayTracker *ttdTracker =
+        [[SentryTimeToDisplayTracker alloc] initForController:controller
+                                           waitForFullDisplay:self.enableWaitForFullDisplay];
+    [vcSpan.tracer addExtension:ttdTracker];
+
+    self.currentTTDTracker = ttdTracker;
+}
+
+- (void)finishTTIDForController:(UIViewController *)controller
+{
+    SentrySpan *vcSpan = [self viewControllerPerformanceSpan:controller];
+
+    if (![vcSpan isKindOfClass:[SentryTracer self]]) {
+        // Only tracer may have TimeToDisplayTracker
+        return;
```

Comment:
How does execution arrive at this spot? The fact that this kind of condition needs to be tested in the `if` statement is a code smell.

Reply:
A controller may be a child of another controller, therefore instead of having a SentryTracer it will have a SentrySpan.

Reply:
Thanks for the explanation, would you mind adding it to the comment here to explain the relationship between spans/tracer and parent/child VCs?

AFAIUI, we start a span for every VC in a hierarchy, then just pick the one for the top level parent to actually use? If that is correct, what happens to the rest of the spans?

The code smell here is the fact that we have to care about the receiving type at all in the implementation. If SentryTracer were not a subclass of SentrySpan, there would be no need for checks, casts, or calls to `-[tracer]`. 

The TTD span should simply be attached to the correct view controller, decided by whatever is managing TTD tracking, which would look at the VC parent/child relationships at the start, so there would only ever be one span.

Reply:
Now its me that is not understanding your explanation 😁, but it does not really matter any more because the approach changed to a simpler one.